src/modules/workspaces/interceptors/workspaces-logs.interceptor.ts

```diff
@@ -0,0 +1,332 @@
+    payload: Omit<WorkspaceLogAttributes, 'id' | 'createdAt' | 'updatedAt'>,
+  ) {
+    try {
+      await this.workspaceRepository.registerLog({
+        ...payload,
+        createdAt: new Date(),
+        updatedAt: new Date(),
+      });
+    } catch (error) {
+      Logger.debug(
+        `[WORKSPACE/LOGS] An error occurred trying to register a log of type ${payload.type} for the user ${payload.creator}`,
+        error,
+      );
+    }
+  }
+
+  async getUserWorkspaces(uuid: string) {
+    const availableWorkspaces =
+      await this.workspaceRepository.findUserAvailableWorkspaces(uuid);
+    return availableWorkspaces
+      .filter(
+        ({ workspace, workspaceUser }) =>
+          workspace.isWorkspaceReady() && !workspaceUser.deactivated,
+      )
+      .map(({ workspace }) => workspace.id);
+  }
+
+  async handleUserAction(
+    platform: WorkspaceLogPlatform,
+    actionType: WorkspaceLogType,
+    req: any,
+    res: any,
+  ) {
+    const user: User = res?.user || req?.user;
+
+    if (!user?.uuid) {
+      Logger.debug('[WORKSPACE/LOGS] User is required');
+      return;
+    }
+
+    const workspaceIds = await this.getUserWorkspaces(user.uuid);
+    await Promise.all(
+      workspaceIds.map((workspaceId) =>
+        this.registerWorkspaceLog({
+          workspaceId,
+          creator: user.uuid,
+          type: actionType,
+          platform,
+        }),
+      ),
+    );
+  }
+
+  async handleUserWorkspaceAction(
+    platform: WorkspaceLogPlatform,
+    actionType: WorkspaceLogType,
+    req: any,
+    res: any,
+    entity?: string,
+  ) {
+    const { ok, requesterUuid, workspaceId } = this.extractRequestData(req);
+
+    const entityId = entity || this.getEntity(req, res);
+    if (!entityId) {
+      Logger.debug('[WORKSPACE/LOGS] Item Id is required');
+    }
+
+    if (ok && requesterUuid && workspaceId && entityId) {
+      await this.registerWorkspaceLog({
+        workspaceId,
+        creator: requesterUuid,
+        type: actionType,
+        platform,
+        entityId,
+      });
+    }
+  }
+
+  getItemType(req: any): string {
+    return req?.body?.itemType || req?.params?.itemType;
+  }
+
+  getEntity(req: any, res: any): string {
+    return req?.body?.itemId || req?.params?.itemId || res?.itemId;
+  }
+
+  determineAction(
+    type: 'SHARE' | 'DELETE',
+    itemType: WorkspaceItemType,
+  ): WorkspaceLogType {
+    const actionMap = {
+      SHARE: {
+        [WorkspaceItemType.File]: WorkspaceLogType.SHARE_FILE,
+        [WorkspaceItemType.Folder]: WorkspaceLogType.SHARE_FOLDER,
+      },
+      DELETE: {
+        [WorkspaceItemType.File]: WorkspaceLogType.DELETE_FILE,
+        [WorkspaceItemType.Folder]: WorkspaceLogType.DELETE_FOLDER,
+      },
+    };
+
+    const action = actionMap[type]?.[itemType];
+
+    if (!action) {
+      Logger.debug(
+        `[WORKSPACE/LOGS] Invalid action type: ${type} or item type: ${itemType}`,
+      );
+      return null;
+    }
+
+    return action;
+  }
```

Comment:
Honestly, this just adds complexity for such a simple function you just used twice. It kinda makes hard to understand what you're trying to accomplish. 

I know you want to encapsulate things, but sometimes jumping between code, specially something just called `determineAction`, forces other devs to try to understand what you are trying to do

Reply:
@apsantiso This was thinking about the future, since you may want to record other logs such as `move` or `rename` which are applied to files and folders. However, for the moment it was not considered for this first approach. In case later on we need to add both here we would not need to create so many if for validation.

Anyway, let me know if I destroy the function and add the number of if separately.

Reply:
If you find too much trouble while trying to remove the function, then try to change its name to something more specific, like `determineItemAction`, so we can know this is something used specifically to determine what action is being used on an item.